pkg/server/stream/v3/stream.go

```diff
@@ -23,20 +23,44 @@ type DeltaStream interface {
 
 // StreamState will keep track of resource state per type on a stream.
 type StreamState struct { // nolint:golint,revive
-	// Indicates whether the original DeltaRequest was a wildcard LDS/RDS request.
+	// Indicates whether the stream currently has a wildcard watch
 	wildcard bool
 
+	// Provides the list of resources explicitly requested by the client
+	// This list might be non-empty even when set as wildcard
+	subscribedResourceNames map[string]struct{}
+
 	// ResourceVersions contains a hash of the resource as the value and the resource name as the key.
 	// This field stores the last state sent to the client.
 	resourceVersions map[string]string
 
 	// knownResourceNames contains resource names that a client has received previously
 	knownResourceNames map[string]map[string]struct{}
 
-	// indicates whether the object has beed modified since its creation
+	// indicates whether the object has been modified since its creation
 	first bool
 }
 
+func (s *StreamState) GetSubscribedResourceNames() map[string]struct{} {
```

Comment:
Can we add some doc strings to these functions? Let's be explicit about delta xds here as well.

Reply:
I added docstring for the three related methods